src/shims/unix/solarish/foreign_items.rs

```diff
@@ -18,6 +19,28 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
     ) -> InterpResult<'tcx, EmulateItemResult> {
         let this = self.eval_context_mut();
         match link_name.as_str() {
+            // Allocation
+            "memalign" => {
+                let [align, size] =
+                    this.check_shim(abi, Abi::C { unwind: false }, link_name, args)?;
+                let align = this.read_target_usize(align)?;
+                let size = this.read_target_usize(size)?;
+
+                // memalign requires the alignment to be a power of 2
+                // and to be, at least, of word size.
+                // http://docs.oracle.com/cd/E88353_01/html/E37743/memalign-3c.html
```

Comment:
This URL shows an error 404 for me.

Reply:
I found it at https://docs.oracle.com/cd/E88353_01/html/E37843/memalign-3c.html.

It doesn't say what happens when the alignment is violated though -- is it UB or a safe error?

Reply:
Ah no it does say something:

> The memalign() function may also fail if:
>
> EINVAL
>
>    The value of the alignment parameter is not a power of two multiple of sizeof(void *), or the size parameter is zero, or the combination of the two would lead to an integer overflow.

But how is the error returned...?
This is where it'd probably be good to experiment with an actual Solaris/Illumos system and see what happens on a bad alignment.^^

Reply:
I just tried on my illumos instance, it returns null and emits EINVAL.

Reply:
What does "emit" mean"?

In [this document](https://open-std.org/JTC1/SC22/WG14/www/docs/summary.htm#dr_460) and in `man memalign` on my system it says that `memalign` returns NULL and sets errno on error.